src/generator.cs

```diff
@@ -1117,6 +1118,145 @@ static string FindProtocolInterface (Type type, MemberInfo pi)
 		return "I" + type.Name;
 	}
 
+	// TODO: Alex
+	public static BindAsAttribute GetBindAsAttribute (ICustomAttributeProvider cu) => GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter);
+	public static bool HasBindAsAttribute (ICustomAttributeProvider cu) => (GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter)) != null;
+	static bool IsSetter (MethodInfo mi) => mi.IsSpecialName && mi.Name.StartsWith ("set_", StringComparison.Ordinal);
+
+	string GetToBindAsWrapper (MemberInformation minfo = null, ParameterInfo pi = null)
+	{
+		BindAsAttribute attrib = null;
+		string className = null;
+		string temp = null;
+
+		if (pi == null) {
+			attrib = GetBindAsAttribute (minfo.mi);
+			var property = minfo.mi as PropertyInfo;
+			var method = minfo.mi as MethodInfo;
+			className = method?.ReturnType?.Name ?? property?.PropertyType?.Name;
+		} else {
+			attrib = GetBindAsAttribute (pi);
+			className = pi.ParameterType.Name;
+		}
+
+		var retType = Nullable.GetUnderlyingType (attrib.Type) ?? attrib.Type;
+		var isNullable = attrib.IsNullable;
+		var isValueType = retType.IsValueType;
+
+		if (isNullable || !isValueType)
+			temp = string.Format ("{0} == null ? null : ", pi != null ? pi.Name.GetSafeParamName () : "value");
+
+		if (className == "NSNumber")
+			temp = string.Format ("new NSNumber ({1}{0});", isNullable ? ".Value" : string.Empty, pi != null ? pi.Name.GetSafeParamName () : "value");
```

Comment:
Added in https://github.com/xamarin/xamarin-macios/pull/1476/commits/a301fdae6041ca52517b54daa1da62bd0f5587e8 thanks for the idea!